### ISSUE TITLE
[BUG] PYMEDataServer name can be too long for later zeroconf versions

### ISSUE THREAD

**Issue by barentine** [python-microscopy/python-microscopy#546]
**Describe the bug**
```
(base) Andrews-MacBook-Pro-5:~ Andrew$ PYMEDataServer
INFO:numexpr.utils:NumExpr defaulting to 8 threads.
INFO:root:========================================
PYMEDataServer, running on python 3.6.10 |Anaconda, Inc.| (default, Jan  7 2020, 15:01:53) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]

INFO:root:Serving from directory: /Users/Andrew/PYMEData
Traceback (most recent call last):
  File "/opt/miniconda3/bin/PYMEDataServer", line 33, in <module>
    sys.exit(load_entry_point('python-microscopy', 'console_scripts', 'PYMEDataServer')())
  File "/Users/Andrew/code/python-microscopy/PYME/cluster/HTTPDataServer.py", line 925, in main
    ns.register_service(service_name, ip_addr, sa[1])
  File "/Users/Andrew/code/python-microscopy/PYME/misc/pyme_zeroconf.py", line 166, in register_service
    desc)
  File "/opt/miniconda3/lib/python3.6/site-packages/zeroconf/__init__.py", line 1773, in __init__
    if not type_.endswith(service_type_name(name, allow_underscores=True)):
  File "/opt/miniconda3/lib/python3.6/site-packages/zeroconf/__init__.py", line 322, in service_type_name
    raise BadTypeInNameException("Too long: '%s'" % remaining[0])
zeroconf.BadTypeInNameException: Too long: 'PYMEDataServer [Andrews-MacBook-Pro-5]: Andrews-MacBook-Pro-5 - PID:883'
(base) Andrews-MacBook-Pro-5:~ Andrew$ conda list | grep zeroconf
zeroconf                  0.28.5                   pypi_0    pypi
```

**To Reproduce**
1. run PYMEClusterOfOne on a laptop with a silly name
2. notice that the dataserver isn't showing up
3. try running the dataserver
4. boom

**Platform (please complete the following information - or copy-paste from error dialog if shown):**
 - OS: [e.g. Win10] osx 15
 - Python version: 3.6
 - numpy version: 1.16
 - wxpython version [if a UI bug]: 4.0.4

**Comment by barentine** [comment]
name needs to be shorter than 64 characters, but even capping all names in pyme_zeroconf.py at 63 chars (saving the PID bit at the end, taking things out of the middle) or more directly just hacking the compName in e.g. HTTPDataServer to be short, then gets us:
```
INFO:root:Serving from directory: /Users/Andrew/PYMEData
Traceback (most recent call last):
  File "/opt/miniconda3/bin/PYMEDataServer", line 33, in <module>
    sys.exit(load_entry_point('python-microscopy', 'console_scripts', 'PYMEDataServer')())
  File "/Users/Andrew/code/python-microscopy/PYME/cluster/HTTPDataServer.py", line 925, in main
    ns.register_service(service_name, ip_addr, sa[1])
  File "/Users/Andrew/code/python-microscopy/PYME/misc/pyme_zeroconf.py", line 181, in register_service
    self.zc.register_service(info)
  File "/opt/miniconda3/lib/python3.6/site-packages/zeroconf/__init__.py", line 2471, in register_service
    self._broadcast_service(info)
  File "/opt/miniconda3/lib/python3.6/site-packages/zeroconf/__init__.py", line 2511, in _broadcast_service
    DNSText(info.name, _TYPE_TXT, _CLASS_IN | _CLASS_UNIQUE, info.other_ttl, info.text), 0
  File "/opt/miniconda3/lib/python3.6/site-packages/zeroconf/__init__.py", line 623, in __init__
    assert isinstance(text, (bytes, type(None)))
AssertionError

```

**Comment by barentine** [comment]
ah, looks like they loosened service name validation in 0.28.6. Will try updating

**Comment by barentine** [comment]
OK, tried that, still hit both those issues

**Comment by barentine** [comment]
As mentioned in #442 , current head only works with zeroconf up to `0.26.3`, however after downgrading I still hit the name too long error (but not the following assertion error if I trim the name)

**Comment by barentine** [comment]
fixed with #548
